src/js/components/hocs.js

```diff
@@ -42,36 +39,32 @@ export const withFocus = (WrappedComponent) => {
 
     componentWillUnmount = () => {
       const { wrappedRef } = this.state;
-      window.removeEventListener('mousedown', this.handleActiveMouse);
       const wrapperNode = findDOMNode(wrappedRef.current);
       if (wrapperNode && wrapperNode.addEventListener) {
         wrapperNode.removeEventListener('focus', this.setFocus);
       }
-      clearTimeout(this.mouseTimer);
-    }
-
-    handleActiveMouse = () => {
```

Comment:
why are we removing the logic of mouse active ?

Reply:
I was encountering issues with keyboard focus setting causing similar timing problems. My thinking is that the primary purpose of `withFocus` is to provide a `focus` property to components that want to render the focus in a certain way. Because of how the focus events traverse components, for example when Select and TextInput have Buttons in their Drops. Both the Buttons and the Select/TextInput are triggering focus events as the user interacts. By using a general timer, not specific to just mouse events, this allows the active element to cycle through elements and none of the grommet components will re-render until the focus has stabilized. Without this, we were seeing cases where changing the `focus` property via `withFocus` was causing a re-render and preventing the delivery of Button events.